### PR TITLE
49 eslint problems reported #173

### LINKED ISSUE #173: 49 eslint problems reported
**Describe the bug**
49 Lint problems (46 errors, 3 warnings) in `pixie-blog` source code.

The majority of the issues are in three categories:
1. Unused imports. Trivial fix.
2. Path resolution. A tweak to the eslint config would most likely fix this.
3. Tiny issues like `==` where there should be a `===`.

**To Reproduce**

1. `nvm exec 13 yarn lint`

**Expected behavior**

There should be no lint problems reported when running `yarn lint`.

**Additional context**

Command:
`$ nvm exec 13 yarn lint`

Output:
```
Running node v13.14.0 (npm v6.14.4)
yarn run v1.22.4
$ eslint src ./*.js; git ls-files '**/*.tsx' '**/*.ts' '**/*.js' '**/*.jsx' '**/*.scss' '**/*.py' | xargs -l1 tools/licenses/checker.py -f

.../pixie-blog/src/components/image-zoom-modal.plugin.js
   99:19  warning  Unexpected unnamed function  func-names
  114:20  warning  Unexpected unnamed function  func-names
  119:21  warning  Unexpected unnamed function  func-names

.../pixie-blog/src/components/mainThemeProvider.tsx
  20:10  error  'useEffect' is defined but never used      no-unused-vars
  20:21  error  'useRef' is defined but never used         no-unused-vars
  22:26  error  'useMediaQuery' is defined but never used  no-unused-vars

.../pixie-blog/src/components/mdxComponents/baseComponents.tsx
  26:28  error  Unable to resolve path to module 'components/utils'     import/no-unresolved
  27:18  error  Unable to resolve path to module './code'               import/no-unresolved
  27:18  error  Missing file extension for "./code"                     import/extensions
  29:23  error  Unable to resolve path to module './anchor'             import/no-unresolved
  29:23  error  Missing file extension for "./anchor"                   import/extensions
  30:26  error  Unable to resolve path to module './codeRenderer'       import/no-unresolved
  30:26  error  Missing file extension for "./codeRenderer"             import/extensions
  31:22  error  Unable to resolve path to module './listItem'           import/no-unresolved
  31:22  error  Missing file extension for "./listItem"                 import/extensions
  32:29  error  Unable to resolve path to module './custom-table-cell'  import/no-unresolved
  32:29  error  Missing file extension for "./custom-table-cell"        import/extensions
  33:19  error  Unable to resolve path to module './h-link'             import/no-unresolved
  33:19  error  Missing file extension for "./h-link"                   import/extensions
  34:23  error  Unable to resolve path to module './footnotes'          import/no-unresolved
  34:23  error  Missing file extension for "./footnotes"                import/extensions

.../pixie-blog/src/components/mdxComponents/codeRenderer.tsx
   25:34  error  Unable to resolve path to module '../mainThemeProvider'  import/no-unresolved
   25:34  error  Missing file extension for "../mainThemeProvider"        import/extensions
  101:28  error  Expected '===' and instead saw '=='                      eqeqeq

.../pixie-blog/src/components/mdxComponents/command.tsx
  20:10  error  'Theme' is defined but never used  no-unused-vars

.../pixie-blog/src/components/mdxComponents/custom-table-cell.tsx
  22:21  error  Unable to resolve path to module './parseMd'           import/no-unresolved
  22:21  error  Missing file extension for "./parseMd"                 import/extensions
  35:1   error  This line has a length of 134. Maximum allowed is 100  max-len

.../pixie-blog/src/components/mdxComponents/footnotes.tsx
  39:1  error  This line has a length of 106. Maximum allowed is 100  max-len

.../pixie-blog/src/components/mdxComponents/h-link.tsx
  36:22  error  'TypographyVariant' is defined but never used  no-unused-vars

.../pixie-blog/src/components/mdxComponents/index.tsx
  21:28  error  Unable to resolve path to module './baseComponents'  import/no-unresolved
  21:28  error  Missing file extension for "./baseComponents"        import/extensions
  22:25  error  Unable to resolve path to module './svg'             import/no-unresolved
  22:25  error  Missing file extension for "./svg"                   import/extensions
  23:21  error  Unable to resolve path to module './command'         import/no-unresolved
  23:21  error  Missing file extension for "./command"               import/extensions
  24:19  error  Unable to resolve path to module './quote'           import/no-unresolved
  24:19  error  Missing file extension for "./quote"                 import/extensions
  25:24  error  Unable to resolve path to module './block-quote'     import/no-unresolved
  25:24  error  Missing file extension for "./block-quote"           import/extensions

.../pixie-blog/src/components/mdxComponents/listItem.tsx
  33:1  error  This line has a length of 102. Maximum allowed is 100  max-len

.../pixie-blog/src/components/mdxComponents/parseMd.tsx
  20:1   error  '@mdx-js/runtime' should be listed in the project's dependencies. Run 'npm i -S @mdx-js/runtime' to add it  import/no-extraneous-dependencies
  21:27  error  Unable to resolve path to module './index'                                                                  import/no-unresolved
  21:27  error  Missing file extension for "./index"                                                                        import/extensions

.../pixie-blog/src/components/mdxComponents/quote.tsx
  22:10  error  'Theme' is defined but never used  no-unused-vars

.../pixie-blog/src/components/mdxComponents/svg.tsx
  21:34  error  Unable to resolve path to module '../mainThemeProvider'  import/no-unresolved
  21:34  error  Missing file extension for "../mainThemeProvider"        import/extensions

.../pixie-blog/src/components/theme.js
  150:8  error  Missing trailing comma  comma-dangle

.../pixie-blog/src/pages/404.tsx
  22:10  error  'Theme' is defined but never used  no-unused-vars

✖ 49 problems (46 errors, 3 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

### PR DESCRIPTION
**Pull request by postescu**
Fixes #173

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,75 +1,91 @@
...
+        "moduleDirectory": [
+          "node_modules",
+          "src/"
+        ]
+      }
+    }
+  },
+  "rules": {
+    "import/extensions": 0,
+    "import/no-unresolved": 1,
```
Comment (vihangm) [inline .eslintrc.json:67]: just delete the line so we inherit the default settings

File: package.json [hunk 2]
```diff
@@ -106,6 +106,7 @@
     "eslint-plugin-promise": "^4.2.1",
     "eslint-plugin-react": "^7.18.3",
     "eslint-plugin-react-hooks": "^1.7.0",
+    "eslint-config-airbnb-typescript": "^7.2.1",
```
Comment (vihangm) [inline package.json:109]: don't we need to add `'airbnb-typescript'` to the eslint plugins section to use this?

**Review by vihangm — CHANGES_REQUESTED** [review]

**Comment by vihangm** [comment]
Can you please include `Fixes #173` in the description (and do that for future PRs too to link to the issue correctly).

File: .eslintrc.json (hunk 1 above)
Comment (postescu) [reply .eslintrc.json]: done

File: package.json (hunk 2 above)
Comment (postescu) [reply package.json:109]: done

File: src/components/mdxComponents/index.tsx [hunk 3]
```diff
@@ -23,11 +23,14 @@ import SvgRenderer from './svg';
...
+import CustomTableCell from './custom-table-cell';
 
 export default {
   ...baseComponents,
   svg: (props: any) => <SvgRenderer {...props} />,
   quote: (props: any) => <Quote {...props} />,
   blockquote: (props: any) => <BlockQuote {...props} />,
   command: (props: any) => <Command {...props} />,
+  td: ({ align, ...props }) => <CustomTableCell {...props} align={align || undefined} />,
+
```
Comment (vihangm) [inline src/components/mdxComponents/index.tsx:35]: why did this move from `baseComponents.tsx`?

Comment (postescu) [reply src/components/mdxComponents/index.tsx:35]: Cyclic dependency issue.

**Review by vihangm — APPROVED** [review]

**Pull request merged**